docs/admin/reference-architectures.md

```diff
@@ -0,0 +1,168 @@
+
+Users can connect to workspaces using SSH or via workspace applications like
+`code-server`, facilitating collaboration and remote access. Additionally,
+workspaces can be parameterized, allowing users to customize settings and
+configurations based on their unique needs. Workspaces are instantiated using
+Coder templates and deployed on nodes by provisioners.
+
+### Template
+
+A template in Coder is a predefined configuration for creating workspaces.
+Templates streamline the process of workspace creation by providing
+pre-configured settings, tooling, and dependencies. They are built by template
+administrators on top of Terraform, allowing for efficient management of
+infrastructure resources. Additionally, templates can utilize Coder modules to
+leverage existing features shared with other templates, enhancing flexibility
+and consistency across deployments. Templates describe provisioning rules for
+infrastructure resources offered by cloud providers.
+
+### Proxy
+
+A workspace proxy serves as a relay connection option for developers connecting
+to their workspace over SSH, a workspace app, or through port forwarding. It
+helps reduce network latency for geo-distributed teams by minimizing the
+distance network traffic needs to travel. Notably, workspace proxies do not
+handle dashboard connections or API calls.
+
+### Provisioner
+
+Provisioners in Coder execute Terraform during workspace and template builds.
+While the platform includes built-in provisioner daemons by default, there are
+advantages to employing external provisioners. These external daemons provide
+secure build environments, and reduce server load, improving performance and
+scalability. Each provisioner can handle a single concurrent workspace build,
+allowing for efficient resource allocation and workload management.
+
+### Registry
+
+The Coder Registry hosts starter templates for various cloud providers and
+orchestration platforms, enabling self-service cloud development environments
+via Terraform-defined infrastructure. Additionally, Coder introduces _Modules_
+to streamline template creation by extracting commonly used functionalities such
+as web IDEs, third-party integrations, and helper scripts into reusable
+components.
+
+The Registry is hosted service and it is not available for air-gapped
+deployments.
+
+### Kubernetes cluster for Coder
+
+A dedicated cluster for Coder is Kubernetes cluster specifically configured to
+host and manage Coder workloads. Kubernetes provides container orchestration
+capabilities, allowing Coder to efficiently deploy, scale, and manage workspaces
+across a distributed infrastructure. This ensures high availability, fault
+tolerance, and scalability for Coder deployments.
+
+The cluster can be deployed using the Helm chart.
+
+## Scale tests methodology
+
+Scaling Coder involves careful planning and testing to ensure it can handle more
+users without slowing down. This process encompasses infrastructure setup,
```

Comment:
```suggestion
Scaling Coder involves careful planning and testing to ensure it can handle more
load without compromising service. This process encompasses infrastructure setup,
```

Reply:
Fixed